src/OpenLoco/src/GameCommands/Road/RemoveRoadStation.cpp

```diff
@@ -0,0 +1,156 @@
+#include "RemoveRoadStation.h"
+#include "Economy/Economy.h"
+#include "Map/RoadElement.h"
+#include "Map/StationElement.h"
+#include "Map/TileElement.h"
+#include "Map/TileManager.h"
+#include "Map/Track/TrackData.h"
+#include "Objects/ObjectManager.h"
+#include "Objects/RoadStationObject.h"
+#include "ViewportManager.h"
+#include "World/Station.h"
+#include "World/StationManager.h"
+
+namespace OpenLoco::GameCommands
+{
+    // TODO: based on CreateRoadStation.cpp
+    static World::RoadElement* getElRoad(World::Pos3 pos, uint8_t rotation, uint8_t roadObjectId, uint8_t roadId, uint8_t index)
+    {
+        auto tile = World::TileManager::get(pos);
+        for (auto& el : tile)
+        {
+            auto* elRoad = el.as<World::RoadElement>();
+            if (elRoad == nullptr)
+            {
+                continue;
+            }
+            if (elRoad->baseHeight() != pos.z)
+            {
+                continue;
+            }
+            if (elRoad->unkDirection() != rotation)
+            {
+                continue;
+            }
+            if (elRoad->roadId() != roadId)
+            {
+                continue;
+            }
+            // TODO: not seeing this check in disasm
+            if (elRoad->roadObjectId() != roadObjectId)
+            {
+                continue;
+            }
+            if (elRoad->sequenceIndex() != index)
+            {
+                continue;
+            }
+            return elRoad;
+        }
+        return nullptr;
+    }
+
+    // 0x0048D2AC
+    static currency32_t removeRoadStation(const RoadStationRemovalArgs& args, const uint8_t flags)
+    {
+        setExpenditureType(ExpenditureType::Construction);
+        setPosition(args.pos + World::Pos3(16, 16, 0));
+        bool updateStationTileRegistration = true;
+
+        // Get road at position
+        auto* initialElRoad = getElRoad(args.pos, args.rotation, args.type, args.roadId, args.index);
+        if (initialElRoad == nullptr)
+        {
+            return FAILURE;
+        }
+
+        // No station element on this road?
+        if (!initialElRoad->hasStationElement())
+        {
+            return FAILURE;
+        }
+
+        // Find the station element for an ownership check
+        {
+            World::StationElement* stationEl = nullptr;
+            while (auto* nextElement = initialElRoad->next())
```

Comment:
although this matches vanilla its not really correct. The station element will be one after the last roadelement. This just looks for the first station element after any road element. Subtly different. I would do something like:
```cpp
// Station element must be the next element after the last consecutive road element
World::StationElement* stationEl = [initialElRoad](){
    auto* elRoad = initialElRoad;
    while(!elRoad->isLast()){
        auto* nextEl = elRoad->next();
        elRoad = nextEl->as<World::RoadElement>();
        if (elRoad != nullptr){
           continue;
        }
        stationEl = nextEl->as<World::StationElement>();
        if (stationEl != nullptr){
            return stationEl;
        }
        return nullptr;
    }
   return nullptr;
}();

Reply:
Ahh i see another thing vanilla does later on that we should add into this.
```
if (elRoad != nullptr && elRoad->baseHeight() == pos.Z){
   continue;
}```